vhdbuilder/packer/install-dependencies.sh

```diff
@@ -243,6 +243,26 @@ downloadCNI() {
     retrycmd_get_tarball 120 5 "$downloadDir/${cniTgzTmp}" ${CNI_PLUGINS_URL} || exit $ERR_CNI_DOWNLOAD_TIMEOUT
 }
 
+downloadAndInstallCriTools() {
+  downloadDir=${1}
+  evaluatedURL=${2}
+  version=${3}
+
+  # if downloadDir and evaluatedURL are not empty, download and install crictl by this override, which is the old way to install
+  if [[ "${downloadDir}" != "" && "${evaluatedURL}" != "" ]]; then
```

Comment:
```suggestion
  if [ "${downloadDir}" != "" ] && [ "${evaluatedURL}" != "" ]; then
```
we should see if we can add our own custom rules to shellcheck

Reply:
let me check

Reply:
Updated for this one.
I found a way to check the POSIX compliant scripts but it caught a lot of errors as expected. I will try to fix all of them in a new PR today.
The command will be changed to this one. `shellcheck "--shell=sh"  $(printf -- "-e %s " $IGNORED) $filesToCheck`